src/ptvsd/_vendored/pydevd/tests_python/test_debugger_json.py

```diff
@@ -2159,6 +2161,135 @@ def test_source_mapping(case_setup):
         writer.finished_ok = True
 
 
+@pytest.mark.skipif(not TEST_CHERRYPY, reason='No CherryPy available')
+def test_process_autoreload_cherrypy(case_setup_multiprocessing, tmpdir):
+    '''
+    CherryPy does an os.execv(...) which will kill the running process and replace
+    it with a new process when a reload takes place, so, it mostly works as
+    a new process connection (everything is the same except that the
+    existing process is stopped).
+    '''
+    port = get_free_port()
+    # We write a temp file because we'll change it to autoreload later on.
+    f = tmpdir.join('_debugger_case_cherrypy.py')
+
+    tmplt = '''
+import cherrypy
+cherrypy.config.update({'server.socket_port': %(port)s})
```

Comment:
Explicit is better than implicit + disable checker so that you won't have to filter out logs.
```suggestion
cherrypy.config.update({
    'checker.on': False,
    'engine.autoreload.on': False,
    'server.socket_port': %(port)s,
})
```